tests/fixtures/expect.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fixtures
+
+import (
+	"context"
+	"testing"
+
+	"github.com/numaproj-labs/numaplane/pkg/apis/numaplane/v1alpha1"
+	planepkg "github.com/numaproj-labs/numaplane/pkg/client/clientset/versioned/typed/numaplane/v1alpha1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Expect struct {
+	t             *testing.T
+	restConfig    *rest.Config
+	kubeClient    kubernetes.Interface
+	gitSync       *v1alpha1.GitSync
+	gitSyncClient planepkg.GitSyncInterface
+}
+
+// check that resources are created
+func (e *Expect) ResourcesExist(resourceType string, resources []string) *Expect {
+
+	e.t.Helper()
+	ctx := context.Background()
+	for _, r := range resources {
+		result := e.kubeClient.CoreV1().RESTClient().Get().
+			Timeout(defaultTimeout).
+			Namespace(TargetNamespace).
+			Resource(resourceType).
+			Name(r).
+			Do(ctx)
+		if result.Error() != nil {
+			e.t.Fatalf("Resource %s does not exist", r)
+		}
+	}
+
+	return e
+}
+
+// check that resources are deleted
+func (e *Expect) ResourcesDontExist(resourceType string, resources []string) *Expect {
+
+	e.t.Helper()
+	ctx := context.Background()
+	for _, r := range resources {
+		result := e.kubeClient.CoreV1().RESTClient().Get().
+			Timeout(defaultTimeout).
+			Namespace(TargetNamespace).
+			Resource(resourceType).
+			Name(r).
+			Do(ctx)
```

Comment:
I don't think this will work. Let's say the call to the K8S API to delete the resources was just called, and then this is called. This will find the resource which hasn't quite been deleted yet and it will return an error. The `defaultTimeout` won't have done its job.